File: app/src/main/java/com/ninhttd/devtest/data/local/TekoDb.java
package com.ninhttd.devtest.data.local;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import com.ninhttd.devtest.BuildConfig;
import com.ninhttd.devtest.TekoApplication;
import com.ninhttd.devtest.data.entity.Product;
import com.ninhttd.devtest.data.local.dao.ProductDao;


@Database(entities = {Product.class}, version = BuildConfig.VERSION_CODE, exportSchema = false)
@TypeConverters(Converter.class)
public abstract class TekoDb extends RoomDatabase {
    public static String DATABASE_NAME = "teko_db";
    private static TekoDb db;

    public static TekoDb getTekoDb() {
        if (db == null) {
            db = Room.databaseBuilder(TekoApplication.getAppContext(), TekoDb.class, DATABASE_NAME)
                    .allowMainThreadQueries()
                    .setJournalMode(JournalMode.TRUNCATE)
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return db;
    }

    public static void deleteDb(Context context) {
        context.deleteDatabase(DATABASE_NAME);
        db = null;
    }

    public abstract ProductDao productDao();
}

File: app/src/main/java/com/ninhttd/devtest/data/local/TekoDb.kt
package com.ninhttd.devtest.data.local

import android.content.Context

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

import com.ninhttd.devtest.BuildConfig
import com.ninhttd.devtest.TekoApplication
import com.ninhttd.devtest.data.entity.ProductEntity
import com.ninhttd.devtest.data.local.dao.ProductDao
import androidx.sqlite.db.SupportSQLiteDatabase
import androidx.room.migration.Migration


@Database(entities = [ProductEntity::class], version = BuildConfig.VERSION_CODE, exportSchema = false)
@TypeConverters(Converter::class)
abstract class TekoDb : RoomDatabase() {

    abstract fun productDao(): ProductDao


    companion object {
        var DATABASE_NAME = "teko_db"
        private var db: TekoDb? = null

        val MIGRATION_1_2: Migration = object : Migration(1, 2) {
            override fun migrate(database: SupportSQLiteDatabase) {
                // Since we didn't alter the table, there's nothing else to do here.
            }
        }

        val tekoDb: TekoDb
            get() {
                if (db == null) {
                    db = Room.databaseBuilder(TekoApplication.appContext!!, TekoDb::class.java, DATABASE_NAME)
                            .addMigrations(MIGRATION_1_2)
                            .setJournalMode(RoomDatabase.JournalMode.TRUNCATE)
                            .fallbackToDestructiveMigration()
                            .build()
                }
                return db as TekoDb
            }

        fun deleteDb(context: Context) {
            context.deleteDatabase(DATABASE_NAME)
            db = null
        }
    }
}
